src/components/barchart/BarChart.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import styled from "styled-components";
+import theme from "../../theme";
+import { space } from "styled-system";
+
+interface IBarValue {
+  color: string;
+  value: number;
+}
+
+interface IBarValues {
+  values: IBarValue[];
+}
+
+interface IStyledBarProps {
+  height?: string;
+}
+
+const StyledBarChart = styled.svg`
+  width: 100%;
+  height: ${(props: IStyledBarProps) =>
+    props.height ? props.height : "1.5rem"};
+  border-radius: ${theme.radiusDefault};
+  ${space};
+`;
+
+type BarProps = IBarValues & IStyledBarProps;
+
+export const BarChart = ({ values, ...visualProps }: BarProps) => {
+  const sum = values.reduce((s, { value }) => s + value, 0);
+  let x = 0;
```

Comment:
If you want to keep mutable value around for calculations, it's better to use useRef
https://www.codementor.io/@dhananjaykumar/useref-hook-as-mutable-ref-object-ykb77fwvk